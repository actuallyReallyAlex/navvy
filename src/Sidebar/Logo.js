import React from 'react'
import PropTypes from 'prop-types'
import { Box, Text } from 'grommet'
import { Dashboard } from 'grommet-icons'

const Logo = ({ size }) => {
  return (
    <Box
      align="center"
      justify="center"
      height="70px"
      style={{ boxShadow: '0 4px 2px -2px gray', minHeight: '70px' }}
    >
      {size === 'expanded' ? (
        <Text weight="bold" margin="none" pad="none">
          D A S H B O A R D
        </Text>
      ) : (
        <Dashboard />
      )}
    </Box>
  )
}

Logo.propTypes = {
  size: PropTypes.oneOf(['expanded', 'collapsed']).isRequired
}

export default Logo
