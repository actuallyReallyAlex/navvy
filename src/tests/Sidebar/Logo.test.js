import React from 'react'
import Logo from '../../Sidebar/Logo'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<Logo />', () => {
  afterEach(() => {})

  test('Should render the <Logo /> component with sidebar expanded.', () => {
    const component = shallow(<Logo {...defaultProps} size="expanded" />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <Logo /> component with sidebar collapsed.', () => {
    const component = shallow(<Logo {...defaultProps} size="collapsed" />)
    expect(component).toMatchSnapshot()
  })
})
