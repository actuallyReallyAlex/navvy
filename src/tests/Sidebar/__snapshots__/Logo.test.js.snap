// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> Should render the <Logo /> component with sidebar collapsed. 1`] = `ShallowWrapper {}`;

exports[`<Logo /> Should render the <Logo /> component with sidebar expanded. 1`] = `ShallowWrapper {}`;
